WordPress/Classes/Utility/ContextManager.m

```diff
@@ -203,95 +204,89 @@ - (void)mergeChanges:(NSManagedObjectContext *)context fromContextDidSaveNotific
-    return _managedObjectModel;
-}
-
-- (NSPersistentStoreCoordinator *)persistentStoreCoordinator
+- (NSPersistentContainer *)persistentContainer
 {
-    if (_persistentStoreCoordinator) {
-        return _persistentStoreCoordinator;
+    if (_persistentContainer) {
+        return _persistentContainer;
     }
-    
-    [self migrateDataModelsIfNecessary];
 
-    // Attempt to open the store
-    _migrationFailed = NO;
-    
-    NSURL *storeURL = self.storeURL;
+    SentryStartupEvent *startupEvent = [SentryStartupEvent new];
 
-    // This is important for automatic version migration. Leave it here!
-    NSDictionary *options = @{
-        NSInferMappingModelAutomaticallyOption            : @(YES),
-        NSMigratePersistentStoresAutomaticallyOption    : @(YES)
-    };
+    [self migrateDataModelsIfNecessary:startupEvent];
 
-    NSError *error = nil;
-    SentryStartupEvent *startupEvent = [SentryStartupEvent new];
-    _persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc]
-                                   initWithManagedObjectModel:[self managedObjectModel]];
-
-    if (![_persistentStoreCoordinator addPersistentStoreWithType:NSSQLiteStoreType
-                                                   configuration:nil
-                                                             URL:storeURL
-                                                         options:options
-                                                           error:&error]) {
-        DDLogError(@"Error opening the database. %@\nDeleting the file and trying again", error);
-
-        SentryStartupEventAddError(startupEvent, error);
-        error = nil;
-
-        _migrationFailed = YES;
-        
-        // make a backup of the old database
-        [[NSFileManager defaultManager] copyItemAtPath:storeURL.path
-                                                toPath:[storeURL.path stringByAppendingString:@"~"]
-                                                 error:&error];
+    NSURL *storeURL = self.storeURL;
 
+    // Initialize the container
+    NSPersistentContainer *persistentContainer = [[NSPersistentContainer alloc] initWithName:@"WordPress" managedObjectModel:self.managedObjectModel];
+    persistentContainer.persistentStoreDescriptions = @[self.storeDescription];
+    [persistentContainer loadPersistentStoresWithCompletionHandler:^(NSPersistentStoreDescription *description, NSError *error) {
         if (error != nil) {
+            DDLogError(@"Error opening the database. %@\nDeleting the file and trying again", error);
+
             SentryStartupEventAddError(startupEvent, error);
             error = nil;
-        }
 
-        // delete the sqlite file and try again
-        [[NSFileManager defaultManager] removeItemAtPath:storeURL.path error:&error];
+            // make a backup of the old database
+            [[NSFileManager defaultManager] copyItemAtPath:storeURL.path
```

Comment:
SQLite actually creates up to three files if journaling is enabled (which it is by default in Core Data). If those journal files stick around when we create a new db, I could see that causing an issue bringing up the store after a migration would fail. In the iterative migrator we sort of account for this in the making of a backup and copying over the migrated file by doing the contents of the directory, not a single file. 

I don't know if it would fix anything, but it wouldn't hurt.

Reply:
@astralbodies Do you mean copying the `-wal` and `-shm` files too?

Reply:
I changed to use `CoreDataIterativeMigrator` backup logic!

Reply:
> @astralbodies Do you mean copying the `-wal` and `-shm` files too?

Yup! We could also turn off journaling as an option.